File: Count Number of Teams.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        vector<pair<int, int>> dp;
        int count = 0;
        for (int num: arr) {
            auto i = lower_bound(dp.begin(), dp.end(), make_pair(num, 0));
            int m = 0;
            for (auto j = dp.begin(); j != i; j++) {
                count += j->second;
                m++;
            }
            dp.insert(i, make_pair(num, m));
        }
        return count;
    }
    int numTeams(vector<int>& rating) {
        int count = countTriplets(rating);
        reverse(rating.begin(), rating.end());
        return count + countTriplets(rating);
    }
};

File: Count Number of Teams.py
class Solution:
    def countTriplets(self, arr):
        dp = []
        count = 0
        for num in arr:
            i = bisect.bisect_left(dp, (num, 0))
            for j in range(i):
                count += dp[j][1]
            dp.insert(i, (num, i))
        return count

    def numTeams(self, rating: List[int]) -> int:
        return self.countTriplets(rating) + self.countTriplets(rating[::-1])
